compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt

```diff
@@ -343,6 +355,71 @@ internal class ComposeSceneMediator(
             }
         }
 
+    fun onScrollEvent(
+        position: DpOffset,
+        velocity: DpOffset,
+        event: UIEvent?,
+        eventKind: TouchesEventKind
+    ) {
+        when (eventKind) {
+            TouchesEventKind.BEGAN -> redrawer.ongoingInteractionEventsCount += 1
+            TouchesEventKind.MOVED -> {}
+            TouchesEventKind.ENDED -> redrawer.ongoingInteractionEventsCount -= 1
+        }
+
+        scene.sendPointerEvent(
+            eventType = PointerEventType.Scroll,
+            pointers = listOf(
+                ComposeScenePointer(
+                    id = PointerId(0),
+                    position = position.toOffset(density),
+                    pressed = false,
+                    type = PointerType.Mouse,
+                )
+            ),
+            scrollDelta = velocity.toOffset(density) * SCROLL_VELOCITY_MULTIPLIER,
+            timeMillis = event.timeMillis,
+            nativeEvent = event,
+            button = event?.button,
+            buttons = PointerButtons(event?.buttonMask ?: 0L),
+            keyboardModifiers = PointerKeyboardModifiers(event?.modifierFlags ?: 0L)
+        )
+    }
+
+    fun onHoverEvent(
+        position: DpOffset,
+        event: UIEvent?,
+        eventKind: TouchesEventKind
+    ) {
+        val eventType = when (eventKind) {
+            TouchesEventKind.BEGAN -> PointerEventType.Enter
+            TouchesEventKind.MOVED -> PointerEventType.Move
+            TouchesEventKind.ENDED -> PointerEventType.Exit
+        }
+
+        scene.sendPointerEvent(
+            eventType = eventType,
+            pointers = listOf(
+                ComposeScenePointer(
+                    id = PointerId(0),
+                    position = position.toOffset(density),
+                    pressed = false,
+                    type = PointerType.Mouse,
+                )
+            ),
```

Comment:
```suggestion
            position = position.toOffset(density),
            type = PointerType.Mouse,
```

This way, `pressed` will calculated by `buttons` by the internal scene logic (it should set correctly, not as `false`).

The same should be done in `onScrollEvent`

Reply:
`onHoverEvent` cannot generate "pressed" state because it designed to only hovers.

Reply:
The `pressed` state should be equal to `buttons.areAnyPressed` state otherwise some code somewhere might work incorrectly (we should carefully add a check for it someday)

Reply:
I removed `button` and `buttons` from the event arguments. Anyway, it should never be pressed in `onHoverEvent` and `onScrollEvent` handler.

Reply:
Can we do any drag with touchpad? If so, does it work as touch down/move/up?

Reply:
Yes, drag gesture will be handled by the old `onTouchEvent` method